cmd/ical-relay/profiles.go

```diff
@@ -64,20 +65,44 @@ func getProfileCalendar(profile datastore.Profile, profileName string) (*ics.Cal
 		var err error
 
 		for i, s := range profile.Sources {
-			if i == 0 {
-				// first source gets assigned to base calendar
-				log.Debug("Loading source ", s, " as base calendar")
-				calendar, err = getSource(s)
-				if err != nil {
+			sourceCacheFilename := conf.Server.StoragePath + "calstore/" + profileName + "-cache-" + helpers.GetMD5Hash(s) + ".ics"
+			log.Debugf("Loading source calendar %s of profile %s", s, profileName)
+			ncalendar, err = getSource(s)
+			if err != nil {
+				// do not apply cache on base64 as they should always be valid
+				if strings.HasPrefix(s, "base64") {
 					return nil, err
 				}
-			} else {
-				// all other calendars only load events
-				log.Debug("Loading source ", s, " as additional calendar")
-				ncalendar, err = getSource(s)
-				if err != nil {
+				// check if history file exists
+				if _, err := os.Stat(sourceCacheFilename); os.IsNotExist(err) {
+					log.Debugf("Source-cache file for source %s in profile %s does not exist!", s, profileName)
 					return nil, err
 				}
+				// load history file and assign it to source calendar variable
+				log.Debugf("Loading cache file %s", sourceCacheFilename)
+				ncalendar, err = helpers.LoadCalFile(sourceCacheFilename)
+				if err != nil {
+					log.Errorln(err)
+					return nil, fmt.Errorf("Error loading history file: %s", err.Error())
+				}
+				usedSourceCache = true
```

Comment:
macht es vllt sinn das Datum des ältesten genutzten Cache elements mit zu übergeben, damit eine Fehlermeldung á la "Der Kalender zeigt ggf. [...] erreicht werden konnte. **Letzte Aktualisierung: \<datetime\>**"

Reply:
wollen wir dann evtl. Source+Timedate anzeigen?
à la "Der Kalender zeigt ggf. [...], da folgende Kalender nicht erreicht werden konnten und stattdessen die letzte erfolgreiche Version verwendet wird:
- source1 (timedate)
- source2 (timedate)"

Reply:
Ja, das macht sinn. Kannst du die ganze Info vllt in den custom error type mit aufnehmen?

Reply:
Jap, genau das war mein Plan :)